src/movies.js

```diff
@@ -1,25 +1,158 @@
+    })
+    return spielbergDramas.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    
+    if (moviesArray.length===0){
+        return 0
+    }   
+    
+    let sum = moviesArray.reduce(function(accum, movie){
+        if (!movie.score){
+            return accum
+        }else {return accum + movie.score}
+    },0)
+    
+    let avg = sum/moviesArray.length;
+    return Math.round(avg*100)/100
+}
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+    
+    let dramaMovies = moviesArray.filter(function(movie){
+        if (movie.genre.includes("Drama")){
+            return movie
+        }
+    })
+    if (dramaMovies.length===0){
+        return 0
+    }
+    sum = dramaMovies.reduce(function(acc, movie){
+        return acc + movie.score
+    },0);
+    let avg = sum/dramaMovies.length
+    return Math.floor(avg*100)/100
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+    let sortedMovies = JSON.parse(JSON.stringify(moviesArray))
+    
+    sortedMovies.sort(function(a,b){
+        if (a.year!==b.year){
+            return a.year-b.year
+        }else {
+          if(!a.title&&!b.title){
+            return a.score-b.score
+          }else if (!a.title) {
+            return -1 
+          } else if (!b.title) {
+            return 1 
+          } else {
+            return a.title.localeCompare(b.title)
+          }
+       }
+    })
+    return sortedMovies
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+    let copy = JSON.parse(JSON.stringify(moviesArray))
+    
+    let movieTitles = copy.map(copy => copy.title)
+    movieTitles.sort(function(a,b){return a.localeCompare(b)})
+    return movieTitles.slice(0,20)
+}
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes(moviesArray) {}
+function turnHoursToMinutes(moviesArray) {
+    let copy = JSON.parse(JSON.stringify(moviesArray))
+    return copy.map(function(movie){
+        const durationParts = movie.duration.split(" ")       
+        let hours = parseInt(durationParts[0]) 
+        let minutes
+        if (durationParts.length===1){
+            minutes = 0
+        }else {minutes = parseInt(durationParts[1])}        
+        
+        const durationInMinutes = hours*60+ minutes
+        return {...movie,
+                duration:durationInMinutes}
+    })    
+}
+
 
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg(moviesArray) {}
+function bestYearAvg(moviesArray) {
+    let moviesByYear
```

Comment:
```suggestion
   const movies = JSON.parse(JSON.stringify(moviesArray))
  let bestYear = 0, bestScore = 0

  // What if the array is empty? Return null
  if(movies.length < 1) return null

  movies.map(movie => movie.year)
    .filter((year, index, self) => self.indexOf(year) === index) // saves only the years
    .sort((a, b) => a - b) // sorts incrementally
    .forEach(year => { // let's evaluate one year at a time

      // Filters the movies with the year in turn
      const moviesPerYear = movies.filter((movie) => movie.year === year)

      // creates a variable for the Acc to register the sum of scores
      let yearScore = moviesPerYear.reduce((acc, movie) => {
        return acc + (movie.score / moviesPerYear.length) // Score average
      }, 0)

      // Adjusts the best score
      bestScore < yearScore
        ? ((bestScore = yearScore), (bestYear = year))
        : void 0
    })

  return `The best year was ${bestYear} with an average score of ${bestScore}`
```

another solution you make a great job solving this on your way :rocket: